### PR TITLE
Make 'make check' more consistent, and better documented, across all our projects

### PR DESCRIPTION
**Pull request by bcrocker15**
Makefile:

Change 'check' target to link directly to 'check_in_container';
move original 'check' actions to new 'check_local' target.

Signed-off-by: Ben Crocker <bcrocker@redhat.com>

### CODE REVIEW

**Comment by bcrocker15** [comment]
recheck

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,47 +4,47 @@ Please follow common guidelines for our projects [here](https://github.com/packi
...
 ## Running packit-service locally
 
 Since packit-service is already a fairly complex system, it's not trivial to
-run it locally. This is why we are running everything in containers.
+run it locally. This is why we run everything in containers.
 
 This repository contains [docker-compose.yml](./docker-compose.yml) for
-[docker-compose](https://github.com/docker/compose). Before you run it, we
-suggest you to open the file and read all the comments.
+[podman-compose](https://github.com/containers/podman-compose). Before you run it, we
```
Comment (jpopelka) [inline CONTRIBUTING.md:46]: Not sure this is a good idea.
We investigated `podman-compose` about a year ago in #703 and it was not a good replacement for `docker-compose` - and there have been no new [releases](https://pypi.org/project/podman-compose/#history) since then.
I've just tried the version packaged in F34 and see some errors, so if you want to play with that please do so in a separate PR. Thanks.

Comment (bcrocker15) [reply CONTRIBUTING.md:46]: Jiri,

Thanks for taking a look.  I think we could say that podman-vs.-docker issues are
somewhat outside the scope of making 'make check' more consistent, so I'm
putting the original text back.

**Comment by bcrocker15** [comment]
I attempted changing the default for 'make check' to
check-in-container.  However, I changed it back to check-local, as
attempting to 'make check-in-container' under Zuul appears to result
in a recursive podman invocation.  The symptom of this recursive
invocation is an error message like this:

/bin/sh: docker: command not found
make: *** [Makefile:34: check_in_container] Error 127
make: *** [Makefile:34: check_in_container] Error 2

The same error occurs in packit,
https://github.com/packit/packit/pull/1230, so I'm going to
change the default back to check-local there, too.

**Comment by bcrocker15** [comment]
recheck

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,34 +4,34 @@ Please follow common guidelines for our projects [here](https://github.com/packi
...
 
 ## Building images locally
 
-For building images SOURCE_BRANCH build arg is required. It allows select which branch should be
-used during automated build and test process, based on deployment stage it was executed from.
+For building images the SOURCE_BRANCH build arg is required. It allows selecting which branch should be
+used during the automated build and test process, based on the deployment stage it was executed from.
 
-Docker file build arg SOURCE_BRANCH is set to:
+The Podman file build arg SOURCE_BRANCH is set to:
```
Comment (csomh) [inline CONTRIBUTING.md:16]:
```suggestion
In the Dockerfiles SOURCE_BRANCH is set to:
```

As there are 3 Dockerfiles in this repo, and there are called `Dockerfile*`, so there removing the "docker" terminology here would create confusion. The right term and name for these files would be "Containerfile".

File: Makefile [hunk 3]
```diff
@@ -23,11 +23,12 @@ check:
 	find . -name "*.pyc" -exec rm {} \;
 	PYTHONPATH=$(CURDIR) PYTHONDONTWRITEBYTECODE=1 python3 -m pytest --color=$(COLOR) --verbose --showlocals --cov=packit_service --cov-report=$(COV_REPORT) $(TEST_TARGET)
 
-test_image: files/install-deps-worker.yaml files/install-deps.yaml files/recipe-tests.yaml
+test-image: files/install-deps-worker.yaml files/install-deps.yaml files/recipe-tests.yaml
```
Comment (csomh) [inline Makefile:26]:
```suggestion
build-test-image: files/install-deps-worker.yaml files/install-deps.yaml files/recipe-tests.yaml
```

**Review by csomh — COMMENTED** [review]
`install-check-deps` and `install-check-in-container-deps` should be also added.

File: CONTRIBUTING.md (hunk 2 above)
Comment (bcrocker15) [reply CONTRIBUTING.md:16]: Made the suggested change.

File: Makefile (hunk 3 above)
Comment (bcrocker15) [reply Makefile:26]: Made the suggested change.

File: Makefile [hunk 4]
```diff
@@ -23,11 +23,12 @@ check:
...
 	PYTHONPATH=$(CURDIR) PYTHONDONTWRITEBYTECODE=1 python3 -m pytest --color=$(COLOR) --verbose --showlocals --cov=packit_service --cov-report=$(COV_REPORT) $(TEST_TARGET)
 
-test_image: files/install-deps-worker.yaml files/install-deps.yaml files/recipe-tests.yaml
+build-test-image: files/install-deps-worker.yaml files/install-deps.yaml files/recipe-tests.yaml
 	$(CONTAINER_ENGINE) build --rm -t $(TEST_IMAGE) -f files/docker/Dockerfile.tests --build-arg SOURCE_BRANCH=$(SOURCE_BRANCH) .
 
-check_in_container:
+check-in-container:
 	@# don't use -ti here in CI, TTY is not allocated in zuul
+	@# Note: once in container, 'make check' (else recursive check-in-container)
```
Comment (csomh) [inline Makefile:31]: I would say this is self-explanatory, so there is no need to comment about it (i.e: The comment doesn't communicate any information that cannot be easily figure out just reading the lines bellow)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -197,14 +197,16 @@ We have multiple test categories within packit-service:
 
 ### Running tests locally
 
-You can run unit and integration tests locally in a container:
+You can run unit and integration tests locally in a container;
+make sure that you have the ansible and podman packages installed,
```
Comment (csomh) [inline CONTRIBUTING.md:201]: Is ansible relly required?

Comment (bcrocker15) [reply CONTRIBUTING.md]: I think ansible is only required by check-inside-openshift, check-inside-openshift-zuul,
and setup-inside-toolbox, so I've removed the mention from CONTRIBUTING.md.

File: Makefile (hunk 4 above)
Comment (bcrocker15) [reply Makefile]: I removed the offending comments from Makefile.

**Review by csomh — APPROVED** [review]
lgtm
